apple/internal/local_provisioning_profiles.bzl

```
@@ -0,0 +1,161 @@
+
+def _provisioning_profile_repository(repository_ctx):
+    system_profiles_path = "{}/Library/MobileDevice/Provisioning Profiles".format(repository_ctx.os.environ["HOME"])
+    repository_ctx.execute(["mkdir", "-p", system_profiles_path])
+    repository_ctx.symlink(system_profiles_path, "profiles")
+
+    repository_ctx.file("BUILD", """
+filegroup(
+    name = "profiles",
+    srcs = glob(["profiles/*.mobileprovision"], allow_empty = True),
+    visibility = ["//visibility:public"],
+)
+
+filegroup(
+    name = "empty",
+    srcs = [],
+    visibility = ["//visibility:public"],
+)
+
+alias(
+    name = "fallback_profiles",
+    actual = "{}",
+    visibility = ["//visibility:public"],
+)
+""".format(repository_ctx.attr.fallback_profiles or ":empty"))
+
+provisioning_profile_repository = repository_rule(
+    environ = ["HOME"],
+    implementation = _provisioning_profile_repository,
+    attrs = {
+        "fallback_profiles": attr.label(
+            allow_files = [".mobileprovision"],
+        ),
+    },
+    doc = """
+This rule declares an external repository for discovering locally installed
+provisioning profiles. This is consumed by `ios_local_provisioning_profile`.
+You can optionally set 'fallback_profiles' to point at a stable location of
+profiles if a newer version of the desired profile does not exist on the local
+machine. This is useful for checking in the current version of the profile, but
+not having to update it every time a new device or certificate is added.
+
+## Example
+
+### In your `WORKSPACE` file:
+
+load("@build_bazel_rules_apple//apple:apple.bzl", "provisioning_profile_repository")
+
+provisioning_profile_repository(
+    name = "local_provisioning_profiles",
+    fallback_profiles = "//path/to/some:filegroup", # Optional profiles to use if one isn't found locally
+)
+
+### In your `BUILD` files (see `local_provisioning_profile` for more examples):
+
+load("@build_bazel_rules_apple//apple:apple.bzl", "local_provisioning_profile")
+
+local_provisioning_profile(
+    name = "app_debug_profile",
+    profile_name = "Development App",
+    team_id = "abc123",
+)
+
+ios_application(
+    name = "app",
+    ...
+    provisioning_profile = ":app_debug_profile",
+)
+""",
+)
+
+def _local_provisioning_profile(ctx):
+    if not ctx.files._local_srcs and not ctx.attr._fallback_srcs:
+        ctx.fail("Either local or fallback provisioning profiles must exist")
+
+    selected_profile_path = "{name}.mobileprovision".format(name = ctx.attr.profile_name or ctx.attr.name)
+    selected_profile = ctx.actions.declare_file(selected_profile_path)
+
+    args = ctx.actions.args()
+    args.add(ctx.attr.profile_name or ctx.attr.name)
+    args.add(selected_profile)
+    if ctx.attr.team_id:
+        args.add("--team-id")
+        args.add(ctx.attr.team_id)
+    args.add("--local-profiles")
+    args.add_all(ctx.files._local_srcs)
+    args.add("--fallback-profiles")
+    args.add_all(ctx.files._fallback_srcs)
```

Should argument name and value be in the same call?
```suggestion
        args.add("--team-id", ctx.attr.team_id)
    args.add_all("--local-profiles", ctx.files._local_srcs)
    args.add_all("--fallback-profiles", ctx.files._fallback_srcs)
```

---

Cool!